runtime/src/main/kotlin/kotlin/native/concurrent/WorkerBoundReference.kt

```diff
@@ -55,12 +57,16 @@ public class WorkerBoundReference<out T : Any>(value: T) {
      */
     val worker: Worker = Worker.current
 
+    private lateinit var cleaner: Cleaner
+
     @ExportForCppRuntime("Kotlin_WorkerBoundReference_freezeHook")
     private fun freezeHook() {
         // If this hook was already run, do nothing.
         if (valueBeforeFreezing == null)
             return
         ptr = createWorkerBoundReference(valueBeforeFreezing!!)
+        @OptIn(ExperimentalStdlibApi::class)
+        cleaner = createCleaner(ptr, ::cleanWorkerBoundReference)
```

Comment:
Function reference is an object too (see https://github.com/JetBrains/kotlin-native/pull/4362#discussion_r498040064)

Reply:
Removed `Cleaner` from `WorkerBoundReference` altogether.